### PR TITLE
[Simulator] Add support for radio behind a simRadio flag.

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: src/simulator/Simulator.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ const Simulator = ({
                 ref={ref}
                 as="iframe"
                 // This needs changing before we remove the flag.
-                src={`https://stage-python-simulator.microbit.org/simulator.html?color=${encodeURIComponent(
+                src={`https://deploy-preview-34--distracted-dubinsky-fd8a42.netlify.app/simulator.html?color=${encodeURIComponent(
```
Comment (microbit-robert) [inline src/simulator/Simulator.tsx:90]: Must revert this before merge.

**Comment by microbit-robert** [comment]
Preview build will be at
https://review-python-editor-next.microbit.org/sim-radio/?flag=simulator

File: src/simulator/SimulatorModules.tsx [hunk 2]
```diff
@@ -87,6 +93,7 @@ const SimulatorModules = ({ simState, ...props }: SimulatorModulesProps) => {
   const [sensors, setSensors] = useState<Record<string, Sensor>>(
     device.sensors
   );
+  console.log(sensors);
```
Comment (microbit-robert) [inline src/simulator/SimulatorModules.tsx:96]: Remove debug.

**Comment by microbit-matt-hillsdon** [comment]
The approach in the sim will change quite a bit from this PR. With me to update to target the alt-radio-support branch.

File: src/device/simulator.ts [hunk 3]
```diff
@@ -127,11 +129,25 @@ export class SimulatorDeviceConnection
       case "state_change": {
         this.state = {
           ...this.state,
-          ...event.data.changes,
+          ...event.data.change,
```
Comment (microbit-matt-hillsdon) [inline src/device/simulator.ts:132]: This is a bug in the switch to the new API also affecting accelerometer.set_range.

**Comment by microbit-matt-hillsdon** [comment]
UI needs more tweaks but I'll follow up with that. Also need to look at capping/discarding messages and what we do on group change.

**Pull request merged**